pkg/arvo/sys/hoon.hoon

```diff
@@ -1861,6 +1913,14 @@
 ++  nl
   |%
   ::                                                    ::
+  :: ???  some tricky type inference stuff
+  :: > !>(~[1 2 3])
+  :: [#t/{@ud @ud @ud $~} q=[1 2 3 0]]
+  :: > !>(`(list @)`~[1 2 3])
+  :: [#t/it(@) q=[1 2 3 0]]
+  :: > !>((le:nl ~[1 2 3]))
+  :: [#t/it(?(?(?(#! @ud) @ud) @ud)) q=[1 2 3 0]]
+  :: so.... it infers the depth of the list or something?
```

Comment:
"Depth of list" is actually part of the type, if that makes sense. ie `[@ud @ud @ud ~]` is a different type from `[@ud ~]`. And wet gates maintain type information, so in cases like these you can see it retained in your output too.